test/functional/shared/resources/application_test.go

```diff
@@ -54,3 +59,169 @@ func Test_Application(t *testing.T) {
+					OutputResources:   []*v20231001preview.ApplicationGraphOutputResource{},
+					Connections: []*v20231001preview.ApplicationGraphConnection{
+						{
+							Direction: &DirectionOutbound,
+							ID:        &backRteID,
+						},
+					},
+				},
+				{
+					ID:                &backRteID,
+					Name:              &backRteName,
+					Type:              &backRteType,
+					ProvisioningState: &ProvisioningStateSuccess,
+					Connections: []*v20231001preview.ApplicationGraphConnection{
+						{
+							Direction: &DirectionInbound,
+							ID:        &backCntrID,
+						},
+					},
+				},
+			},
+		},
+	}
+
+	test := shared.NewRPTest(t, name, []shared.TestStep{
+		{
+			Executor: step.NewDeployExecutor(template, functional.GetMagpieImage()),
+			RPResources: &validation.RPResourceSet{
+				Resources: []validation.RPResource{
+					{
+						Name: name,
+						Type: validation.ApplicationsResource,
+					},
+					{
+						Name: "http-front-ctnr-simple1",
+						Type: validation.ContainersResource,
+						App:  name,
+					},
+					{
+						Name: "http-back-rte-simple1",
+						Type: validation.HttpRoutesResource,
+						App:  name,
+					},
+					{
+						Name: "http-back-ctnr-simple1",
+						Type: validation.ContainersResource,
+						App:  name,
+					},
+				},
+			},
+			K8sObjects: &validation.K8sObjectSet{
+				Namespaces: map[string][]validation.K8sObject{
+					appNamespace: {
+						validation.NewK8sPodForResource(name, "http-front-ctnr-simple1"),
+						validation.NewK8sPodForResource(name, "http-back-ctnr-simple1"),
+						validation.NewK8sServiceForResource(name, "http-back-rte-simple1"),
+					},
+				},
+			},
+			PostStepVerify: func(ctx context.Context, t *testing.T, ct shared.RPTest) {
+				// Verify the application graph
+				options := shared.NewRPTestOptions(t)
+				client := options.ManagementClient
+				require.IsType(t, client, &clients.UCPApplicationsManagementClient{})
+				appManagementClient := client.(*clients.UCPApplicationsManagementClient)
+				appGraphClient, err := v20231001preview.NewApplicationsClient(appManagementClient.RootScope, &aztoken.AnonymousCredential{}, appManagementClient.ClientOptions)
+				require.NoError(t, err)
+				res, err := appGraphClient.GetGraph(ctx, "corerp-application-simple1", map[string]any{}, nil)
+				require.NoError(t, err)
+
+				sort.Slice(expectedGraphResp.ApplicationGraphResponse.Resources, func(i, j int) bool {
+					return *expectedGraphResp.ApplicationGraphResponse.Resources[i].ID < *expectedGraphResp.ApplicationGraphResponse.Resources[j].ID
+				})
+
+				sort.Slice(res.ApplicationGraphResponse.Resources, func(i, j int) bool {
+					return *res.ApplicationGraphResponse.Resources[i].ID < *res.ApplicationGraphResponse.Resources[j].ID
+				})
+
+				//iterate over each of the element in res.ApplicationGraphResponse.Resources and compare with expectedGraphResp.ApplicationGraphResponse.Resources
+				for i, resResource := range res.ApplicationGraphResponse.Resources {
```

Comment:
OK. I am adding the output resources to expected and then just use the require.Equals. 
I had omitted that since it was making the object look really heavy.

Reply:
complete